lib/HALTarget/IMU.h

```diff
@@ -0,0 +1,197 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  IMU implementation
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ * 
+ * @addtogroup HALSim
+ *
+ * @{
+ */
+#ifndef IMU_H
+#define IMU_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IIMU.h"
+#include "Zumo32U4IMU.h"
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+  
+/** The IMU adapter. */
+class IMU : public IIMU
+{
+public:
+    /**
+     * Constructs the IMU adapter.
+     *
+     */
+    IMU() :
+        IIMU(),
+        m_rawAccelerometerOffsetX(0),
+        m_rawAccelerometerOffsetY(0),
+        m_rawAccelerometerOffsetZ(0),
+        m_rawGyroOffsetX(0),
+        m_rawGyroOffsetY(0),
+        m_rawGyroOffsetZ(0)
+    {
+    }
+
+    /**
+     * Destroys the IMU adapter.
+     */
+    ~IMU()
+    {
+    }
+
+    template <typename T> struct vector
```

Comment:
What is this for? (Documentation missing.)

Reply:
Resolved since the Vector has been replaces with an IMUData struct.